### PR TITLE
Refactor BYOC to Kubernetes and Secret types

### PR DESCRIPTION
**Pull request by tarokkk**

### CODE REVIEW

File: api/supported.go [hunk 1]
```diff
@@ -35,12 +35,8 @@ func GetSupportedClusterList(c *gin.Context) {
...
 			},
 			{
-				Name: "Build Your Own Cluster",
-				Key:  constants.BYOC,
-			},
-			{
-				Name: "Dummy cluster",
-				Key:  constants.Dummy,
+				Name: "Kubernetes Cluster",
+				Key:  constants.Kubernetes,
```
Comment (baluchicken) [inline api/supported.go]: Are you sure you want to remove the Dummy cluster?

File: cluster/common.go [hunk 2]
```diff
@@ -164,9 +164,9 @@ func CreateCommonClusterFromRequest(createClusterRequest *bTypes.CreateClusterRe
 
 		return dummy, nil
 
-	case constants.BYOC:
+	case constants.Kubernetes:
```
Comment (baluchicken) [inline cluster/common.go]: Wrong comment

File: cluster/common.go [hunk 3]
```diff
@@ -164,9 +164,9 @@ func CreateCommonClusterFromRequest(createClusterRequest *bTypes.CreateClusterRe
...
 		// Create BYOC struct
-		byoc, err := CreateBYOCClusterFromRequest(createClusterRequest, orgId)
+		byoc, err := CreateKubernetesClusterFromRequest(createClusterRequest, orgId)
```
Comment (baluchicken) [inline cluster/common.go]: rename it to kubernetesCluster

File: cluster/kubernetes.go [hunk 4]
```diff
@@ -161,30 +165,30 @@ func (b *BYOCluster) GetAPIEndpoint() (string, error) {
...
 }
 
-// CreateBYOCClusterFromModel converts ClusterModel to BYOCluster
-func CreateBYOCClusterFromModel(clusterModel *model.ClusterModel) (*BYOCluster, error) {
+// CreateBYOCClusterFromModel converts ClusterModel to Kubecluster
+func CreateKubernetesClusterFromModel(clusterModel *model.ClusterModel) (*Kubecluster, error) {
 	log := logger.WithFields(logrus.Fields{"action": constants.TagGetCluster})
 	log.Debug("Create ClusterModel struct from the request")
-	byocCluster := BYOCluster{
+	byocCluster := Kubecluster{
```
Comment (baluchicken) [inline cluster/kubernetes.go]: rename byocCluster

File: api/supported.go (hunk 1 above)
Comment (tarokkk) [reply api/supported.go]: yes unless you want to promote it on the UI

**Pull request merged**